discord/webhook.py

```diff
@@ -504,7 +504,7 @@ def from_url(cls, url, *, adapter):
             The URL is invalid.
         """
 
-        m = re.search(r'discordapp.com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)
+        m = re.search(r'discord.com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)
```

Comment:
this should be changed to make `app` optional, not to require the new URL.

Reply:
As per the announcement channel in Discord API
> We currently plan to start requiring the new domain later this year for third party developers on November 7th, 2020. We will be sending out more formal notices (system DMs and account emails) to announce this breaking change in the coming weeks.

Reply:
Just because Discord is gonna be breaking old URLs doesn't mean we have to do the same.

Reply:
Make it `app` optional until November 7th, 2020, which at that point force users to use `discord.com` and not `discordapp.com`.

Reply:
To put a time on `app` being there is to create a breaking change. It does not matter when. You have to consider people who set-and-forget bots with perfectly fine, yet old webhooks in their configs.

Reply:
All webhooks that are `discordapp.com` also work on the new `discord.com` domain, so any webhook ID/Token that is a valid webhook at `discordapp.com` will also be valid at `discord.com`

Reply:
Not if they're not parsed here, they won't.

Reply:
In this case, could you not just add regex to check for either one?
```py
m = re.search(r'(discord|discordapp).com/api/webhooks/(?P<id>[0-9]{17,21})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})', url)```